### ISSUE TITLE
ERROR: InitError: MethodError: Cannot `convert` an object of type Umlaut.Tape{ONNX.ONNXCtx} to an object of type Ghost.Tape

### ISSUE THREAD

**Issue by zznidar** [sergio-santos-group/ProtoSyn.jl#46]
When I run `using ProtoSyn`, I get the following error: 
```
ERROR: InitError: MethodError: Cannot `convert` an object of type 
  Umlaut.Tape{ONNX.ONNXCtx} to an object of type 
  Ghost.Tape
```

I have just freshly installed Julia 1.8.3, added https://github.com/sergio-santos-group/ProtoSyn.jl.git with Pkg, then in the .julia/conda directory pip-installed torch and torchani as per the warning ProtoSyn prints on the first run. 

<details>
  <summary>Stacktrace</summary>
  
```
 | Loading TorchANI
 | Loading ONNX models
ERROR: InitError: MethodError: Cannot `convert` an object of type 
  Umlaut.Tape{ONNX.ONNXCtx} to an object of type 
  Ghost.Tape
Closest candidates are:
  convert(::Type{T}, ::T) where T at Base.jl:61
  Ghost.Tape(::C) where C at C:\Users\username\.julia\packages\Ghost\WYs14\src\tape.jl:220
Stacktrace:
 [1] convert(#unused#::Type{Union{Nothing, Ghost.Tape}}, x::Umlaut.Tape{ONNX.ONNXCtx})
   @ Base .\some.jl:36
 [2] ProtoSyn.Calculators.GB.GBModels(C::Umlaut.Tape{ONNX.ONNXCtx}, N::Umlaut.Tape{ONNX.ONNXCtx}, H::Umlaut.Tape{ONNX.ONNXCtx}, O::Umlaut.Tape{ONNX.ONNXCtx}, S::Umlaut.Tape{ONNX.ONNXCtx})
   @ ProtoSyn.Calculators.GB C:\Users\username\.julia\packages\ProtoSyn\21g26\src\Core\Calculators\gb.jl:9
 [3] top-level scope
   @ none:1
during initialization of module GB
```
</details>

If I run `using ProtoSyn` again, it doesn't throw an error, however, the `energy_function(pose)` (from the [Example 4](https://github.com/sergio-santos-group/ProtoSyn.jl/blob/master/examples/4-energy-calculation.ipynb)) then throws an error `TypeError: in keyword argument models, expected ProtoSyn.Calculators.GB.GBModels, got a value of type Nothing`, which I suppose has to do something with not having been able to initialise the module GB in the first try.

**Comment by JosePereiraUA** [comment]
Hi @zznidar.
There seems to be an issue when loading the GB models. You can always not include the GB energy function components, but Generalized Born solvation energy is an important energy component.
I've never attempted to run ProtoSyn.jl on Julia 1.8.3, there might be an issue with that. I'll try it soon.
Another option is a problem with the ONNX.jl package itself. Could you check what version of ONNX.jl you have installed?

I'll look into this and try to replicate the problem. Thank you.

**Comment by zznidar** [comment]
Hey,
According to Pkg, ONNX wasn't even added. I have now added it (ONNX v0.2.1), but the error still persists when I run `using ProtoSyn`.

**Comment by JosePereiraUA** [comment]
I'm currently using ONNX v0.2.0, I'm not sure if that makes a difference, but it doesn't hurt to try and downgrade, for the test.
I'll try to update on my end and see if that re-creates the bug.

**Comment by zznidar** [comment]
Bingo! Downgrading ONNX to v0.2.0 solved the error! Thanks for your help, I'll close the issue :)

**Comment by JosePereiraUA** [comment]
I've managed to re-create the bug after updating to v0.2.1 of ONNX.jl.
I'll try to fix the issue within v0.2.1 (which was actually released 15 days ago). If I'm unable to fix this, I'll fixate the requirements for ProtoSyn.jl on v0.2.0 of ONNX.jl.

In the meanwhile, I believe downgrading to this version should fix the issue.

**Comment by JosePereiraUA** [comment]
For future reference, the breaking changes are introduced here: https://github.com/FluxML/ONNX.jl/commit/536dfe317552d091e9ad0b2e36e7c63a22adf7f7

The fix is introduced by changing the Ghost.jl dependency to Umlaut.jl v0.4.8.
I'll introduce this changes in the [dev](https://github.com/sergio-santos-group/ProtoSyn.jl/tree/dev) branch of ProtoSyn.jl.
